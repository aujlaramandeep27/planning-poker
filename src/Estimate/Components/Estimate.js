import React from 'react';
import { connect } from 'react-redux';
import { estimateStory } from '../Actions/EstimateActions';
import { viewHost, viewMeetings } from '../../Navigation/route-actions';
import { Page } from '../../Common/Header';

export function Estimate({ story, currentMeeting, user, goToMeetings, goToHost, estimateStory }) {
  // TODO: Move estimations to store
  const estimation = [1, 2, 3, 5, 8, 13];
  return (
    <Page title={currentMeeting.name} onBack={ goToMeetings }>
      <HostButton hasHost={ currentMeeting && !!currentMeeting.host } onHostClick={ goToHost }/>
      <StoryDescription { ...story } />
      <div className="uk-align-center uk-width-1-1@m">
        { estimation.map((estimate, i) =>
          <button key={ i } onClick={ () => estimateStory(user._id, story.storyId, estimate) }
                  className={ `uk-button uk-margin-small-top uk-width-1-1 uk-inline pp-button ${ story.estimate === estimate ? 'selected' : '' }` }>
            { estimate }
            <span
              className={ 'uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right' }
              hidden={ story.estimate !== estimate }>Selected</span>
          </button>)
        }
      </div>
    </Page>
  );
}

function StoryDescription({ storyId, storyDescription }) {
  return (<dl className="uk-description-list">
    <dt># { storyId } :</dt>
    <dd>{ storyDescription }</dd>
  </dl>);
}

function HostButton({ hasHost, onHostClick }) {
  if (hasHost) {
    return <></>
  } else {
    return <button className="uk-button uk-button-primary uk-button-small uk-position-small uk-position-top-right"
                   onClick={ onHostClick }>Host</button>;
  }
}

function mapStateToProps(state) {
  return {
      story: state.estimateStory,
      currentMeeting: state.currentMeeting,
      user: state.user
  }
}

function mapDispatchToProps(dispatch) {
  return {
      goToMeetings: () => dispatch(viewMeetings()),
      goToHost: () => dispatch(viewHost()),
      estimateStory: (userId, storyId, estimate) => dispatch(estimateStory(userId, storyId, estimate))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Estimate)
