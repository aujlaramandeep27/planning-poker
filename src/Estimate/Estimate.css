.selected-button {
    background-color: var(--secondary-color);
    color: #fff;
}

.estimate-button {
    height: 50px;
    background-color: var(--primary-color);
    color: #fff;
}

.estimate-button:hover {
    background-color: var(--secondary-color);
    color: #fff;
}

.default-button {
    height: 50px;
}
