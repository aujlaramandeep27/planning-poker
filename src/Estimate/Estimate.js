import React from 'react';
import {connect} from "react-redux";
import {viewHost, viewMeetings} from "../Navigation/route-actions";
import {estimateStory} from "./Actions/EstimateActions";

export function Estimate({goToMeetings, goToHost, estimateStory, storyId, storyDescription, estimate}) {

    const estimation = [1,2,3,5,8,13];
    return(
        <div className="default-background">
            <div className="uk-container uk-text-center@m">
                <h1 className="uk-heading-divider uk-margin-top">Estimate</h1>
                <div className="uk-align-center uk-width-1-2@m">  
                    <span onClick={goToMeetings} data-uk-icon="icon: arrow-left; ratio: 3" className="uk-position-large uk-position-top-left"></span>
                    <button className="uk-button uk-button-primary uk-button-small uk-position-small uk-position-top-right" onClick={goToHost}>Host</button>
                    <dl className="uk-description-list">
                        <dt># {storyId} :</dt>
                        <dd>{ storyDescription }</dd>
                    </dl>
                    <div className="uk-align-center uk-width-1-1@m">
                        { estimation.map((number,i) =>
                            <button key={i} onClick={() => estimateStory(number, storyId)} className={`uk-button uk-margin-small-top uk-width-1-1 uk-inline default-button ${estimate === number ? 'selected-button' : 'estimate-button'}`} >
                                {number}
                                <span className={"uk-position-center-right uk-background-muted uk-text-emphasis uk-label uk-margin-small-right"} hidden={estimate !== number}>Selected</span>
                            </button>)
                        }
                    </div>
                </div>
            </div>
      </div>
    );
}

function mapStateToProps(state) {
    return {
        storyId: state.story.storyId,
        storyDescription: state.story.storyDescription,
        estimate: state.story.estimate
    }
}

function mapDispatchToProps(dispatch) {
    return {
        goToMeetings: () => dispatch(viewMeetings()),
        goToHost: () => dispatch(viewHost()),
        estimateStory: (estimate, storyId) => dispatch(estimateStory(estimate, storyId))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Estimate)