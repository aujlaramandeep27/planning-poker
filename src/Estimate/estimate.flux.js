
export const ESTIMATE_STORY = 'ESTIMATE_STORY';

const initialState = {
    storyId: 'Story - ABC',
    storyDescription: 'Describes the story',
    estimate: 0
};

export function estimateStory(estimate, storyId){
    return {
        type: ESTIMATE_STORY,
        payload: { estimate, storyId }
    }
}

export default function storyReducer(state = initialState, action) {
    switch (action.type) {
        case ESTIMATE_STORY:
            return {
                ...state,
                ...action.payload
            };

        default:
            return state;
    }
}