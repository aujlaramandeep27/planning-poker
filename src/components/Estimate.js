import React from 'react';

export default function Estimate({ estimates }) {
  return estimates.map((estimate, index) =>
    <div className="uk-margin" key={index}>
      <button className="uk-button uk-button-default uk-width-1-1">
        {estimate}
      </button>
    </div>
  );
}
