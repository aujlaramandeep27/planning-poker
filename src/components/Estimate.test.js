import Estimate from './Estimate';

describe('Component: Estimate', () => {
  test('should render', () => {
    expect(Estimate({estimates: ['A', 'B', 'C']})).toMatchSnapshot();
  });
});
