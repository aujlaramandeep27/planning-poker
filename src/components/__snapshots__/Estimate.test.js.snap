// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Estimate should render 1`] = `
Array [
  <div
    className="uk-margin"
  >
    <button
      className="uk-button uk-button-default uk-width-1-1"
    >
      A
    </button>
  </div>,
  <div
    className="uk-margin"
  >
    <button
      className="uk-button uk-button-default uk-width-1-1"
    >
      B
    </button>
  </div>,
  <div
    className="uk-margin"
  >
    <button
      className="uk-button uk-button-default uk-width-1-1"
    >
      C
    </button>
  </div>,
]
`;
